### PR TITLE
osx_defaults: add expand_user_and_vars flag to write

### LINKED ISSUE #5234: `~` gets evaluated when using `community.general.osx_defaults`
### Summary

When passing `~` as value to the `community.general.osx_defaults` plugin, it gets directly evaluated to `/Users/pavelzwerschke`. 

This is probably because the value [is not passed](https://github.com/ansible-collections/community.general/blob/82d69bb871c9b8b61423090af54783f0ae78a189/plugins/modules/system/osx_defaults.py#L307) with quotes to the command line.

When doing it manually, we get the same problem:
```console
$ defaults write com.googlecode.iterm2 PrefsCustomFolder ~/.config/iterm2
$ defaults read com.googlecode.iterm2 PrefsCustomFolder
/Users/pavelzwerschke/.config/iterm2
```

Instead, we want ansible to do something like this:
```console
$ defaults write com.googlecode.iterm2 PrefsCustomFolder '~/.config/iterm2'
$ defaults read com.googlecode.iterm2 PrefsCustomFolder
~/.config/iterm2
```

Also, when executing the playbook twice, it still says `changed` instead of `ok` the second time because the string comparison is still evaluated to `false`.

### Issue Type

Bug Report

### Component Name

community.general.osx_defaults

### Ansible Version

```console (paste below)
$ ansible --version
ansible [core 2.13.3]
  config file = /Users/pavelzwerschke/Projekte/ansible-provision/ansible.cfg
  configured module search path = ['/Users/pavelzwerschke/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /opt/homebrew/Caskroom/mambaforge/base/envs/ansible/lib/python3.10/site-packages/ansible
  ansible collection location = /Users/pavelzwerschke/.ansible/collections:/usr/share/ansible/collections
  executable location = /opt/homebrew/Caskroom/mambaforge/base/envs/ansible/bin/ansible
  python version = 3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:41:22) [Clang 13.0.1 ]
  jinja version = 3.1.2
  libyaml = True
```

### Community.general Version

```console (paste below)
$ ansible-galaxy collection list community.general
# /Users/pavelzwerschke/.ansible/collections/ansible_collections
Collection        Version
----------------- -------
community.general 5.5.0

# /opt/homebrew/Caskroom/mambaforge/base/envs/ansible/lib/python3.10/site-packages/ansible_collections
Collection        Version
----------------- -------
community.general 5.5.0
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed
DEFAULT_HOST_LIST(/Users/pavelzwerschke/Projekte/ansible-provision/ansible.cfg) = ['/Users/pavelzwerschke/Projekte/ansible-provision/inventory']
```

### OS / Environment

macOS 12.5.1

### Steps to Reproduce

```yaml (paste below)
---
- hosts: all
  tasks:
    - name: Ensure iTerm2 custom preferences location is set.
      community.general.osx_defaults:
        domain: com.googlecode.iterm2
        key: PrefsCustomFolder
        value: "~/.config/iterm2"
        type: string
        state: present

```

### Expected Results

Expected:
```console
$ defaults read com.googlecode.iterm2 PrefsCustomFolder
~/.config/iterm2
```

### Actual Results

```console (paste below)
$ defaults read com.googlecode.iterm2 PrefsCustomFolder
/Users/pavelzwerschke/.config/iterm2
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by pavelzw**
##### SUMMARY

Fixes #5234

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

osx defaults

##### ADDITIONAL INFORMATION

cf #5234

### CODE REVIEW

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Can you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment)? Thanks.

File: changelogs/fragments/5243-osx-defaults-expand-user-flags.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - osx_defaults - When passing `~` as the value of osx_defaults, it got expanded directly (https://github.com/ansible-collections/community.general/issues/5234). Fix this behavoir by setting expand_user_and_vars to False (https://github.com/ansible-collections/community.general/pull/5243).
```
Comment (felixfontein) [inline changelogs/fragments/5243-osx-defaults-expand-user-flags.yml:2]:
```suggestion
  - osx_defaults - no longer expand ``~`` in ``value`` to the user's home directory, or expand environment variables (https://github.com/ansible-collections/community.general/issues/5234, https://github.com/ansible-collections/community.general/pull/5243).
```

**Comment by pavelzw** [comment]
Thanks for your help @felixfontein!

**Review by felixfontein — APPROVED** [review]
LGTM. Will merge in a few days if nobody complains.

**Comment by felixfontein** [comment]
@pavelzw thanks again for your contribution!

**Pull request merged**